### DISCUSSION TITLE
Changes from 6.1.26 to 6.1.28 that could cause crackling

### DISCUSSION THREAD

**Discussion by Cippo95** [audinux/fedora-spec#33] [General]
Hello and thanks for this repository,

With Fedora kernels (6.2.x) I had a lot of crackling on the microphone and audio in general.
I have found that installing your real time mao kernel 6.1.26 fixed my problems.  
Today I have received the 6.1.28 and it seems that crackling is back.
I guess kernel developers backported something that is causing the problem.  

I'm a noob about the kernel, I'm asking if you have any clue about what change could be causing this.
Just asking, if you have no clue it is fine, as said I'm a noob about this and I don't know where to look.

Thanks!

**Comment by ycollet** [comment]
Thanks for the report.
I had this problem with 5.15. And that's why I stopped building 5.15.
I will wait for a new release of 6.2-rt and check on my side.
In the mean time, maybe you can test the Fedora kernel and add the following parameter to the boot command: "preempt=full"
I will post a message once I rebuilt a new kernel and the tests are OK on my side.

**Comment by ycollet** [comment]
The solution I found with 5.15 was with pure jack. I had crackling when I first start jack. So I stopped jack and then restarted jack. And the second time; the crackling were gone.

**Comment by Cippo95** [comment]
Thanks for answering.

I have previously tested Fedora's stock kernels also with the `preempt=full` flag, as it was described in the performance tuning of Pipewire:  
https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Performance-tuning  

I'm not using my computer to record professionally but I like to do videos on YouTube.
I use OBS for recording and I'm currently using Pipewire (in particular using JACK inputs in OBS).  

I have tested many kernels, using Bodhi repositories for default ones and Copr for vanillas.
Audio was bad with every kernel but 6.1.26 that has very occasional cracks (this is acceptable from my point of view and usage).

I don't really know if the problem lies in Pipewire or the kernel, I have already reported to Pipewire's GitLab the issue, reporting to kernel developers seems a little tricky because they want a vanilla kernel, without any stuff applied (I have a NVIDIA card and proprietary driver) and they communicate with mails... it's doable but I think it is the last resort.
I'm pretty sure I have also tested without the NVIDIA drivers installed and it was the same.

I'll probably give pure JACK a look in the following days, as said I'm not really used to this but I'm learning.  

I don't think my computer is so strange: it has a Ryzen 5 3500X, a popular MSI board (Gaming PLUS Max) and a RTX 2060 Super.
  
I will report again when I have more news, I will stick to kernel 6.1.26 for my recordings and do some research in the free time.  

Thanks again

**Comment by Cippo95** [comment]
I have found out my issue and it is an ADB (Android Debug Bridge) bug making my whole audio glitching:  
https://issuetracker.google.com/issues/150888445  
ADB seems to periodically open USB devices, but this makes whole audio go nuts (not just USB).

OBS was launching ADB because I used a plugin called DroidCam OBS to use my phones as a camera.  
To add the insult to injury the ADB process wouldn't close when I closed OBS: it was pretty difficult to track down.

For some reason kernel 6.1.26 gives me clean recordings with occasional cracks even with this bug; meanwhile 6.1.28 was behaving as any other kernel giving lot of crackling.  

Someone says ADB behaves badly with NVIDIA's USB C controller (hardware capability, I don't even have the port itself).  
I have disabled it as said here:
https://www.reddit.com/r/linuxquestions/comments/qlrcfj/comment/j1fnp4b/

Still I had some audio glitch and I had this sort of glitches at least one year ago with a GTX 1060 (which shouldn't have this controller).  

Right now I think that ADB is just a bad actor (or the Linux kernel doesn't really like what it does).  

Thank you again for your time and for maintaining this kernel, I'll close this discussion with this comment.

**Reply by ycollet** [reply]
Excellent, thankls a lot for all these links.
I have built 6.3.3-rt15. I will put it on COPR soon.

**Comment by ycollet** [comment]
I am still waiting for a new version for 6.3. 6.3.3 has a filesystem corruption problem.

**Comment by ycollet** [comment]
There is an update for 6.1. I will build this one tonight.
